lib/src/error_tracking/dart_exception_processor.dart

```diff
@@ -0,0 +1,335 @@
+  }
+
+  static String _extractModule(Frame frame) {
+    return frame.uri.pathSegments
+        .sublist(0, frame.uri.pathSegments.length - 1)
+        .join('/');
+  }
+
+  static String? _extractFileName(Frame frame) {
+    return frame.uri.pathSegments.isNotEmpty
+        ? frame.uri.pathSegments.last
+        : null;
+  }
+
+  static String _extractAbsolutePath(Frame frame) {
+    // For privacy, only return filename for local file paths
+    if (frame.uri.scheme != 'dart' &&
+        frame.uri.scheme != 'package' &&
+        frame.uri.pathSegments.isNotEmpty) {
+      return frame.uri.pathSegments.last; // Just filename for privacy
+    }
+
+    // For dart: and package: URIs, full path is safe
+    return frame.uri.toString();
+  }
+
+  /// Extracts the module name from the first stack frame
+  /// This is more accurate than guessing from exception type
+  static String? _getExceptionModule(StackTrace stackTrace) {
+    try {
+      final chain = Chain.forTrace(stackTrace);
+
+      // Get the first frame from the first trace (where exception was thrown)
+      if (chain.traces.isNotEmpty && chain.traces.first.frames.isNotEmpty) {
+        final firstFrame = chain.traces.first.frames.first;
+        return _extractModule(firstFrame);
+      }
+    } catch (e) {
+      // If stack trace parsing fails, return null
+    }
+
+    return null;
+  }
+
+  /// Extracts the package name from the first stack frame
+  /// This is more accurate than guessing from exception type
+  static String? _getExceptionPackage(StackTrace stackTrace) {
+    try {
+      final chain = Chain.forTrace(stackTrace);
+
+      // Get the first frame from the first trace (where exception was thrown)
+      if (chain.traces.isNotEmpty && chain.traces.first.frames.isNotEmpty) {
+        final firstFrame = chain.traces.first.frames.first;
+        return _extractPackage(firstFrame);
+      }
+    } catch (e) {
+      // If stack trace parsing fails, return null
+    }
+
+    return null;
+  }
+
+  /// Gets the current thread ID using isolate-based detection
+  static int _getCurrentThreadId() {
+    try {
+      // Check if we're in the root isolate (main thread)
+      if (isolate_utils.isRootIsolate()) {
+        return 'main'.hashCode;
+      }
+
+      // For other isolates, use the isolate's debug name
+      final isolateName = isolate_utils.getIsolateName();
+      if (isolateName != null && isolateName.isNotEmpty) {
+        return isolateName.hashCode;
+      }
+
+      // Fallback for unknown isolates
+      return 1;
+    } catch (e) {
+      // Graceful fallback if isolate detection fails
+      return 1;
+    }
+  }
+
+  static String _getExceptionType(dynamic error) {
+    // For primitives (String, int, bool, double, null, etc.), just use "Error"
+    if (_isPrimitive(error)) {
+      return 'Error';
+    }
+
+    return error.runtimeType.toString();
+  }
+
+  /// Checks if a value is a primitive type
+  static bool _isPrimitive(dynamic value) {
```

Comment:
i'd avoid the use of `dynamic` everywhere, always prefer Object, so we know if its nullable or not